### PR TITLE
feat(frontend/shared): pronouns on feed page tm-344

### PR DESCRIPTION
**Pull request by KrasKa2**
Screenshot 2024-03-08 at 17 44 12

### CODE REVIEW

File: apps/frontend/src/modules/activities/libs/helpers/get-activity-title/get-activity-title.helper.ts [hunk 1]
```diff
@@ -3,20 +3,25 @@ import {
...
+import { UserSex, UserSexToPronoun } from "~/modules/users/users.js";
 
 const getActivityTitle = (
 	activity: ActivityResponseDto<ValueOf<typeof ActivityType>>,
 ): string => {
 	const userFullName = `${activity.user.firstName} ${activity.user.lastName}`;
 	const { title } = activity.payload;
 
+	// or it is better to set default value in UserAuthResponseDto and UserEntity?
+	const sex = activity.user.sex || UserSex.PREFER_NOT_TO_SAY;
```
Comment (fshabanov) [inline apps/frontend/src/modules/activities/libs/helpers/get-activity-title/get-activity-title.helper.ts:15]:
```suggestion
	const sex = activity.user.sex ?? UserSex.PREFER_NOT_TO_SAY;
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Review by Ann-Maksay — APPROVED** [review]

File: packages/shared/src/modules/users/libs/maps/user-sex-to-pronoun.map.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { UserSex } from "../enums/enums.js";
+
+const UserSexToPronoun = {
```
Comment (what1s1ove) [inline packages/shared/src/modules/users/libs/maps/user-sex-to-pronoun.map.ts:3]: > A map is a very useful and frequently used data structure in any programming language. For maps in the JavaScript world, there is a specific naming convention. The name should follow the pattern aToB, where a serves as the key for retrieving values from the map, followed by the preposition To, implying the mapping of two things, and then B, which is the mapped value for a.

https://dev.to/what1s1ove/javascript-naming-conventions-are-important-45n

File: apps/frontend/src/modules/activities/libs/helpers/get-activity-title/get-activity-title.helper.ts [hunk 3]
```diff
@@ -3,20 +3,24 @@ import {
...
+import { UserSex, UserSexToPronoun } from "~/modules/users/users.js";
 
 const getActivityTitle = (
 	activity: ActivityResponseDto<ValueOf<typeof ActivityType>>,
 ): string => {
 	const userFullName = `${activity.user.firstName} ${activity.user.lastName}`;
 	const { title } = activity.payload;
 
+	const sex = activity.user.sex ?? UserSex.PREFER_NOT_TO_SAY;
+	const pronoun = UserSexToPronoun[sex].OBJECTIVE;
```
Comment (what1s1ove) [inline apps/frontend/src/modules/activities/libs/helpers/get-activity-title/get-activity-title.helper.ts:15]:
```suggestion
	const { sex } = activity.user;
	const pronoun = UserSexToPronoun[sex ?? UserSex.PREFER_NOT_TO_SAY].OBJECTIVE;
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: packages/shared/src/modules/users/libs/maps/user-sex-to-pronoun.map.ts (hunk 2 above)
Comment (what1s1ove) [reply packages/shared/src/modules/users/libs/maps/user-sex-to-pronoun.map.ts:3]: is not resolved

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Comment by what1s1ove** [comment]
@KrasKa2 resolves conflicts please

**Review by GEOFARL — APPROVED** [review]

**Pull request merged**